src/mame/tvgames/xavix_2002.cpp

```diff
@@ -346,13 +346,36 @@ void xavix_i2c_jmat_state::xavix2002_i2c_jmat(machine_config &config)
 	m_xavix2002io->write_2_callback().set(FUNC(xavix_i2c_jmat_state::write_extended_io2));
 }
 
-void xavix2002_superpttv_state::xavix2002_superpctv(machine_config& config)
+DEVICE_IMAGE_LOAD_MEMBER(xavix2002_super_tv_pc_state::cart_load)
+{
+	if (!image.loaded_through_softlist())
+		return std::make_pair(image_error::UNSUPPORTED, "Cartridges must be loaded from the software list");
+
+	uint32_t size = m_cart->common_get_size("prg");
+
+	if (size != 0x400000)
+		return std::make_pair(image_error::UNSUPPORTED, "must be 0x400000 in size");
+
+	m_cart->rom_alloc(size, GENERIC_ROM16_WIDTH, ENDIANNESS_BIG);
+	m_cart->common_load_rom(m_cart->get_rom_base(), size, "prg");
+	memcpy(memregion("bios")->base(), m_cart->get_rom_base(), size);
+
+	return std::make_pair(std::error_condition(), std::string());
+}
```

Comment:
Since this only needs a single ROM region and doesn’t use `feature` elements, is there a reason not to support loading loose files?

Also the 16-bit big endian region is somewhat suspect given the Xavix CPUs are 8-bit little endian.

Reply:
I'd rather things just be added to the softlists, I consider loose-loading to be an anti-feature in cases like this.

I think the SuperXaviX are still 8-bit, although they do use 16-bit data bus for the ROMs (which is one way you can usually tell if something is SuperXaviX, there are 16 data pins on the ROM rather than 8, but it seems transparent to the operation otherwise)

Reply:
> I'd rather things just be added to the softlists, I consider loose-loading to be an anti-feature in cases like this.

Loose software loading is always useful for people developing homebrew and doing A/B testing with Flash carts on real hardware to debug emulation.

> I think the SuperXaviX are still 8-bit, although they do use 16-bit data bus for the ROMs (which is one way you can usually tell if something is SuperXaviX, there are 16 data pins on the ROM rather than 8, but it seems transparent to the operation otherwise)

The trouble is, MAME requires the region to match the CPU’s address space, or you need shim functions to translate, and the CPU cores have 8-bit address spaces in MAME.  (Kind of similar to the situation where Virtual Boy needs 32-bit regions for what’s actually 16-bit ROM, and the mess with WonderSwan which has dynamic bus sizing that MAME can’t emulate efficiently.)

Reply:
I've changed it to 8 anyway, even if I have no confidence either way.

Reply:
You’re still using a 16-bit big endian intermediate region here.